tensorflow/java/src/main/java/org/tensorflow/types/Types.java

```diff
@@ -0,0 +1,37 @@
+package org.tensorflow.types;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.tensorflow.DataType;
+
+/**
+ * Utility class for representing TF types as Java types. For each TF type (e.g., int32),
+ * there is a corresponding Java type (e.g., TFInt32) that represents it at compile time
+ * and a corresponding immutable object (e.g., TFInt32.T) that represents it at run time.
+ */
+public class Types {
+
+  private Types() {} // not instantiable
+
+  static final Map<Class<?>, Integer> typeCodes = new HashMap<>();
+  /** Convert to the equivalent DataType. */
+
+  public static DataType dataType(Class<?> c) {
+    Integer code = typeCodes.get(c);
+    if (code == null) {
+      throw new IllegalArgumentException("" + c + " is not a TensorFlow type.");
+    }
+    return DataType.fromC(code.intValue());
```

Comment:
Wouldn't this require `fromC` to be public? (Since the package for `DataType` is `org.tensorflow`, which is different from the package for this class - `org.tensorflow.types`)?

Alternatively, would it make sense for the map to be:

```
static final Map<Class<?>, DataType> typeMap = new HashMap();
```

instead? (And the `.csv` file can refer to enums in the `DataType`  instead of integers).
So the call in the generated files would look like:

```
static {
  Types.typeCodes.put(TFUint8.class, DataType.UINT8);
}
```

Or perhaps I'm missing something?